src/cpp/include/decoder.h

```diff
@@ -7,6 +7,7 @@
 
 #include "batch.h"
 #include "datatypes.h"
+#include "loss.h"
```

Comment:
Can this include statement be moved to `decoder.cpp`? It's best to keep includes out of the header files if they aren't needed in the header.

Reply:
It seems this is still needed in the header file when declearing `LossFunction *loss_function_;`